internal/x/os/path.go

```diff
@@ -10,7 +10,18 @@ import (
 	"os"
 )
 
-func CleanupTempDir(dir string) error {
+type PathManager interface {
```

Comment:
I think "Managers" are a strong design smell, as they are buckets of utilities functions with unclear boundaries and responsibilities (ie: when we don't know how to call something, it becomes a manager). In this case I suspect the implementation arose from the need of a test, but I would probably go the other way around and create a dedicated set of files and folders for that single test instead of spreading this implementation around the codebase so to support that one use case. That doesn't mean a Filesystem abstraction isn't useful per se, but here it doesn't seem the cost is justified, and the abstraction should probably be refined a bit more, as there are a lot of places where we still use `os` functions straight away.

Reply:
reverted and added a dedicated set of files